include/envoy/network/listener.h

```diff
@@ -278,6 +285,31 @@ class UdpListenerCallbacks {
    * UdpListenerCallback
    */
   virtual UdpPacketWriter& udpPacketWriter() PURE;
+
+  /**
+   * Returns the id of this worker, in the range of [0, concurrency).
+   */
+  virtual uint32_t workerId() const PURE;
```

Comment:
From an interface perspective it seems sub-optimal to pass around both concurrency and worker ID. Presumably you need this because you want stable buckets? I wonder though if this could be more like the TCP version where these details are internal to the packet routing implementation based on what is registered? Like it seems like internally there could be a vector and a stable routing mechanism?

Reply:
In a future change, I think I'm going to need the kernel and the worker to agree on what id the worker has (the kernel's notion of it is defined by the order that sockets in the SO_REUSEPORT group are bound), and using the workerId to encode information into new ConnectionIDs.

Reply:
> In a future change, I think I'm going to need the kernel and the worker to agree on what id the worker has (the kernel's notion of it is defined by the order that sockets in the SO_REUSEPORT group are bound), and using the workerId to encode information into new ConnectionIDs.

Right now the order is undefined, right? Are you going to have to fix the other issue we have been talking about off and on where we pre-create all the sockets on the main thread before starting up the listener on the workers?

So, pedantically, is this going to be an opaque ID or an index? We used to pass around an "id" and then at one point switched to an opaque "name". If this is an index that is < concurrency, can we s/id/index to make it more clear? Also, would it be better to combine the "name" and "index" into a WorkerId struct and pass that around?

Reply:
Yes, I'll need to fix something to have sockets bind in defined order, either by doing on main thread, or some terrible hack where each worker posts to the next worker after it finishes.

Agree, `index` is a better name here; that's how I always plan to use it, and it should always be < concurrency.